pkg/scheduler/flavorassigner/tas_flavorassigner.go

```diff
@@ -68,10 +76,21 @@ func podSetTopologyRequest(psAssignment *PodSetAssignment,
 		return nil, errors.New("workload requires Topology, but there is no TAS cache information for the assigned flavor")
 	}
 	podSet := &wl.Obj.Spec.PodSets[podSetIndex]
+	var podSetUpdates []*kueue.PodSetUpdate
+	for _, ac := range wl.Obj.Status.AdmissionChecks {
+		if ac.State == kueue.CheckStateReady {
```

Comment:
Is this check necessary? Don't we do the second pass when all check are ready?

Reply:
yeah, it may not be necessary indeed, I think this was added proactively, but is consistent with the second pass when all checks ready. 

So, I was thinking that maybe one day we don't schedule second pass when all ready, but as soon as ProvReq is ready. However, indeed, I think for now it is not necessary.